File: Python/employeeEx.py
from company import Company
from person import Person

class EmployeeEx(Person):
    id              = int
    jobPosition     = str
    company         = Company("", 0)    
    startDate       = str
    endDate         = str

    def __init__(self, jobPosition, company, name, surName, nuip):
        super().__init__(name, surName, nuip)
        self.jobPosition = jobPosition
        self.company = company
        


    def printEmployee(self):
        print(self.jobPosition)
        print(vars(self.company))
        print(self.name)

File: Java/EmployeeEx.java
import java.util.Date;

public class EmployeeEx extends Person {
    Integer id;
    String jobPosition;
    Date startDate;
    Date endDate;    
    Company company;

    public EmployeeEx(
        String jobPosition, Company company, 
        String name, String surName, Integer nuip){

        super(name, surName, nuip);
        this.jobPosition = jobPosition;        
        this.company = company;
    }

    public void printEmployee(){
        System.out.println("Person Name: " + super.name);
        System.out.println("    Company     : " + company.bussinessName);
        System.out.println("    Job Position: " + this.jobPosition);
    }
}
